### DISCUSSION TITLE
Improvents to deploy Pimcore to production Server

### DISCUSSION THREAD

**Discussion by idna001** [pimcore/pimcore#16905] [General]
Hey, 
I have several Pimcore installations. But I'm still figuring out how to deploy my changes to my production Server. 
I'm new to Symfony Projects, so I just wanted to ask and start a discussion of ways to deploy. 

This is my current bare stadium: 

1. Code stored in GitHub 
2. GitHub Action running on every Merge
3. Starting an FTP deployment (using [SamKirkland/FTP-Deploy-Action](https://github.com/SamKirkland/FTP-Deploy-Action)@4.2.0)

This is mostly, working fine, but I want to add some things.
Here are my missing points: 

1. I have to often to clear the cache manually 
2. Dependencies won't be affected - if I change them, I do this manually
3. Pimcore Configurations need manual care if I change things in the backend

Mostly my worries are about problem 1 and 2. 
I just want to have stable CI, even for little changes. 
So it's an agile concept

Did I forget something? Are there any examples of which I can get more information about my issues? 
Or is there even a better way to deploy a Pimcore Project?

Thanks for some tips
Andreas

**Comment by dpfaffenbauer** [comment]
Did you read this? https://pimcore.com/docs/platform/Pimcore/Deployment/

**Reply by idna001** [reply]
Yes of course. Since there is no example, I already understood that there is no one solution. 
I just wanted to see some other opinions or ways to deploy. I'm sure there are better ways for that. And I would like to learn from more experienced programmers... 

I posted this here and especially not on a symfony thread as I might get some other solutions in Pimcores deployment, of which I am not aware today.

**Reply by dpfaffenbauer** [reply]
You can use this for example: https://deployer.org/docs/7.x/recipe/pimcore

Or go a containerised approach as lots of others do as well.

**Reply by idna001** [reply]
Okay, thank you, I will take a look at the deployer you mentioned. 
This seems like a good approach to add it as GitHub Action.
